socketio/src/client/callback.rs

```diff
@@ -4,35 +4,39 @@ use std::{
 };
 
 use super::Client;
-use crate::Payload;
+use crate::{Event, Payload};
 
-type InnerCallback = Box<dyn for<'a> FnMut(Payload, Client) + 'static + Sync + Send>;
+pub(crate) type SocketCallback = Box<dyn for<'a> FnMut(Payload, Client) + 'static + Sync + Send>;
+pub(crate) type SocketAnyCallback =
+    Box<dyn for<'a> FnMut(Event, Payload, Client) + 'static + Sync + Send>;
 
-pub(crate) struct Callback {
-    inner: InnerCallback,
+pub(crate) struct Callback<T> {
+    inner: T,
 }
```

Comment:
I don't know if being generic here is the right way.... It's not to be expected that we have any other kind of callback. In that case an enum is usually the way to go, but that is difficult with the deref impl, maybe we find a way around? Otherwise we could just have to callback structs.

Reply:
I tought socket.io server implementation would also use different callbacks.

Reply:
Yeah that is true, but probarbly of the same type. Anyhow, lets use this for a second that way.